contracts/InterchainTokenService.sol

```diff
@@ -932,11 +947,9 @@ contract InterchainTokenService is
         string calldata destinationChain,
         bytes memory destinationAddress,
         uint256 amount,
-        uint32 metadataVersion,
+        MetadataVersion metadataVersion,
         bytes memory data
     ) internal {
-        if (metadataVersion > LATEST_METADATA_VERSION) revert InvalidMetadataVersion(metadataVersion);
```

Comment:
can we keep this check? I would like to revert explicitly. Maybe cache the max enum in a constant?